lib: location: Fallback event and more data details (#13894)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

  * Added:

>     * The :c:enumerator:`LOCATION_EVT_STARTED` event to indicate that location request has been started, and
>       the :c:enumerator:`LOCATION_EVT_FALLBACK` event to indicate that a fallback from one location method to another has occurred.

Reviewer: ```suggestion
    * The :c:enumerator:`LOCATION_EVT_STARTED` event to indicate that a location request has been started and
      the :c:enumerator:`LOCATION_EVT_FALLBACK` event to indicate that a fallback from one location method to another has occurred.
```

      These are for metrics collection purposes and sent only if the :kconfig:option:`CONFIG_LOCATION_DATA_DETAILS` Kconfig option is set.
    * Support for multiple event handlers.
    * Additional location data details into the :c:struct:`location_data_details` structure hierarchy.

>     * The :c:enumerator:`LOCATION_METHOD_WIFI_CELLULAR` method, which cannot be added into the location configuration given to location_request() function,
>       but may occur within the :c:struct:`location_event_data` structure.

Reviewer: ```suggestion
    * The :c:enumerator:`LOCATION_METHOD_WIFI_CELLULAR` method that cannot be added into the location configuration passed to the :c:func:`location_request()` function,
      but may occur within the :c:struct:`location_event_data` structure.
```
